provider/datagen/src/bin/datagen/args.rs

```diff
@@ -0,0 +1,372 @@
+                    .collect::<Result<_, eyre::Error>>()?,
+            )
+        })
+    }
+
+    fn make_cldr(&self) -> eyre::Result<config::PathOrTag> {
+        Ok(match (&self.cldr_root, self.cldr_tag.as_str()) {
+            (Some(path), _) => config::PathOrTag::Path(path.clone()),
+            #[cfg(feature = "networking")]
+            (_, "latest") => config::PathOrTag::Latest,
+            #[cfg(feature = "networking")]
+            (_, tag) => config::PathOrTag::Tag(String::from(tag)),
+            #[cfg(not(feature = "networking"))]
+            _ => eyre::bail!("--cldr-root flag is mandatory unless datagen is built with the `\"networking\"` feature"),
+        })
+    }
+
+    fn make_icu_export(&self) -> eyre::Result<config::PathOrTag> {
+        Ok(match (&self.icuexport_root, self.icuexport_tag.as_str()) {
+            (Some(path), _) => config::PathOrTag::Path(path.clone()),
+            #[cfg(feature = "networking")]
+            (_, "latest") => config::PathOrTag::Latest,
+            #[cfg(feature = "networking")]
+            (_, tag) => config::PathOrTag::Tag(String::from(tag)),
+            #[cfg(not(feature = "networking"))]
+            _ => eyre::bail!("--icuexport-root flag is mandatory unless datagen is built with the `\"networking\"` feature"),
+        })
+    }
+
+    fn make_exporter(&self) -> eyre::Result<config::Export> {
+        match self.format {
+            v @ (Format::Dir | Format::DeprecatedDefault) => {
+                if v == Format::DeprecatedDefault {
+                    log::warn!("Defaulting to --format=dir. This will become a required parameter in the future.");
+                }
+                #[cfg(not(feature = "provider_fs"))]
+                eyre::bail!("FsDataProvider export requires the provider_fs Cargo feature.");
+                #[cfg(feature = "provider_fs")]
+                Ok(config::Export::Fs {
+                    output_path: if let Some(root) = self.output.as_ref() {
+                        root.clone()
+                    } else {
+                        PathBuf::from("icu4x_data")
+                    },
+                    syntax: match self.syntax {
+                        Syntax::Bincode => config::FsSyntax::Bincode,
+                        Syntax::Postcard => config::FsSyntax::Postcard,
+                        Syntax::Json if self.pretty => config::FsSyntax::JsonPretty,
+                        Syntax::Json => config::FsSyntax::Json,
+                    },
+                    fingerprint: self.fingerprint,
+                })
+            }
+            Format::Blob => {
+                #[cfg(not(feature = "provider_blob"))]
+                eyre::bail!("BlobDataProvider export requires the provider_blob Cargo feature.");
+                #[cfg(feature = "provider_blob")]
+                Ok(config::Export::Blob(if let Some(path) = &self.output {
+                    path.clone()
+                } else {
+                    PathBuf::from("/stdout")
```

Comment:
Question: how does this `/stdout` work? Is it portable?

Reply:
It's just a magic constant so that we don't have to introduce an enum between stdout and a file path in the `Config`. I don't think stdout is a useful enough feature to complicate the config schema, and it can be achieved with setting the out path to `/dev/stdout` on Unix at least.

Reply:
I want to keep this an undocumented implementation detail just to pipe the CLI functionality through `Config`.